### DISCUSSION TITLE
SDK build for 32 bit system target

### DISCUSSION THREAD

**Discussion by kronus-lx** [aws/aws-sdk-cpp#2545] [Q&A] [answered]
I'm looking to build the SDK targeting 32-bit applications. So far I have managed to successfully build the SDK however when attempting to use it to build 32-bit binaries/libs my development environment recognizes it as a 64-bit binary and therefore incompatible. I've used various cmake arguments to attempt to resolve this but I keep getting the same result.

**Comment by yasminetalby** [comment]
Hello @kronus-lx ,

Thank you very much for reaching out. 
Could you please provide the following information:
- AWS CPP SDK version used
- Compiler and Version used
- Operating System and version
- CMake invocation

Thank you for your time and collaboration.

Sincerely,

Yasmine

**Reply by kronus-lx** [reply]
Hello,

Certainly, please see below:

- Version: 1.11.105
- MSVC 19.35.32217.1
- Windows 10 Enterprise 22H2
- cmake version 3.26.3

Thank you for your time and early awaiting your response

Kind Regards

Joel Manning

**Reply by yasminetalby** [reply]
Hello Joel,

Thank you very much for your answer. I sincerely apologize if there was any confusion in my question, by Cmake invocation I did not mean cmake version but rather how are you building the SDK using Cmake? you have mentioned using various cmake arguments, could you please provide these details?

Best,

Yasmine

**Reply by kronus-lx** [reply]
Hello Yasmine,

Ahh yes my apologies. 

I have used for instance the following commands:

 - cmake ../aws-sdk-cpp -G "Visual Studio 17 2022" -A Win32 -DBUILD_ONLY=s3
- cmake ../aws-sdk-cpp -G "Visual Studio 17 2022" -A x86 -DBUILD_ONLY=s3

And from there using MSBuild:

MSBuild.exe ./ALL_BUILD -p:Configuration=Debug -p:Plateform=Win32 

Kind  Regards

Joel Manning

**Reply by midget214** [reply]
We are running into the exact same issue.  So far nothing seems to be generating the project files correctly

**Comment by jmklix** [comment, marked as answer]
We currently do not support building this sdk on a 32 bit system. Please make a feature request if you would like us to add that feature.
